### ISSUE TITLE
Should we define a guideline rule for vector math heavy algorithms?

### ISSUE THREAD

**Issue by RedBlight** [OpenGenus/cosmos#3541]
**This is a(n):**

- [ ] New algorithm
- [ ] Update to an existing algorithm
- [ ] Error
- [x] Proposal to the Repository

**Details:**
How are we supposed to handle vector math in our code? I have a cool vector class but it is ~800 LOC. Should I directly paste it on top of all of my vector-heavy mathematical algorithms?

Or, should I include it as a seperate file?

Should I upload it somewhere in the repository only once, and then include from there in all my algorithms?

What are we supposed to do in these kind of situations, where we have to include same ~1000 LOC in many many different algorithms? Perhaps a guideline entry would be an improvement.

**Comment by AdiChat** [comment]
Yes, we must set guidelines to define actions in such a scenario. :+1: 

Keeping all code in one file or the same folder reduces its dependencies to other sections.

Uploading the common file once in a major section for use in sub-sections makes more sense of uniformity to a user exploring that particular section. This should be our preferred path. 

For this, we need to:

* Define a rule in [Cosmos Guides](https://github.com/OpenGenus/cosmos/tree/master/guides) :book: explaining this situation with an example

From the point of contributions, we must:
* Place the common code in the root folder of the concerned section
* Encourage contributors to use the common code in the concerned algorithm

In fact, you can start following this to set an example in your latest contribution. :muscle: 

Feel free to go forward implementing/ setting this rule. :+1:

**Comment by RedBlight** [comment]
For now then I'll just add my "LuVector.hpp" in the folder where "algorithm.cpp" is in, and then include .hpp in the .cpp files, until we decide on something later.
